src/contrib/mlir/backend/pass/affine_lowerer.cpp

```diff
@@ -537,9 +603,548 @@ namespace
+                                                 LLVM::Linkage::Internal,
+                                                 builder.getZeroAttr(globalType),
+                                                 builder);
+            auto globalPtr = builder.create<LLVM::AddressOfOp>(builder.getUnknownLoc(), globalVal);
+
+            switch (m_attrsTyVec[i])
+            {
+            case AttrsType::INT:
+            {
+                auto castOp = builder.create<LLVM::BitcastOp>(
+                    builder.getUnknownLoc(),
+                    getLLVMType(AttrsType::INT, llvmDialect).getPointerTo(),
+                    globalPtr);
+                auto intOp = builder.create<LLVM::ConstantOp>(
+                    builder.getUnknownLoc(), llvmI64Ty, builder.getI64IntegerAttr(attrs.intAttr));
+                builder.create<LLVM::StoreOp>(builder.getUnknownLoc(), intOp, castOp);
+                break;
+            }
+            case AttrsType::GEMM:
+            {
+                auto gemmTy = getLLVMType(AttrsType::GEMM, llvmDialect);
+                auto castOp = builder.create<LLVM::BitcastOp>(
+                    builder.getUnknownLoc(), gemmTy.getPointerTo(), globalPtr);
+
+                std::vector<LLVM::GEPOp> geps;
+                std::vector<LLVM::LLVMType> elemsTy{llvmI8Ty,
+                                                    llvmI8Ty,
+                                                    llvmI64Ty,
+                                                    llvmI64Ty,
+                                                    llvmI64Ty,
+                                                    llvmI64Ty,
+                                                    llvmI64Ty,
+                                                    llvmI64Ty,
+                                                    llvmF32Ty,
+                                                    llvmF32Ty,
+                                                    llvmI32Ty};
+                for (auto j = 0; j < maxNumElem; j++)
+                {
+                    auto gepGemmOp =
+                        builder.create<LLVM::GEPOp>(builder.getUnknownLoc(),
+                                                    elemsTy[j].getPointerTo(),
+                                                    castOp,
+                                                    ArrayRef<Value>({constants[0], constants[j]}));
+                    geps.push_back(gepGemmOp);
+                }
+                // Store attribute values
+                createStore(
+                    llvmI8Ty,
+                    builder.getI8IntegerAttr(static_cast<int8_t>(attrs.gemmAttrs2d.transposeA)),
+                    geps[0],
+                    builder);
+                createStore(
+                    llvmI8Ty,
+                    builder.getI8IntegerAttr(static_cast<int8_t>(attrs.gemmAttrs2d.transposeB)),
+                    geps[1],
+                    builder);
+                createStore(
```

Comment:
llvm type and Attribute passed to createStore are not the same for all. I can combine the same types into a loop. Otherwise, we need these two lists: one for llvm types like {llvmI8Ty, ...} and one for Attributes like {builder.getI8IntegerAttr(static_cast<int8_t>(attrs.gemmAttrs2d.transposeA)), ...}.